### PR TITLE
Demo 3: Optimization Based Planning

### PR DESCRIPTION
**Pull request by mpowelson**
This pull request adds Demo 3, Optimization Based Planning. This demo takes the student through the process of doing PCL perception and pathplanning using TrajOpt and Tesseract on the KUKA iiwa 7kg. The end result is a pick and place workcell using either a suction cup or a Schunk gripper. The process proceeds as follows

1. There are some introductory sections covering setup. The student brings the system up, but perception and pathplanning are broken
2. The student uses PCL to build a perception system that combines data from 3 simulated realsense cameras to detect the top of a box placed in the environment. The system now finds the top of the box but does not pathplan
3. An introduction to TrajOpt is given. The student fills some helper functions that are used to create a TrajOpt problem. The system will now perform a simulated pick and place operation.
4. The student may now optionally move to hardware. The tutorial goes through the steps to set up a KUKA iiwa.
5. The student may now choose to either integrate a Schunk WSG050 gripper or a suction cup controlled via an Acromag 951ELN-4012 GPIO module and a solenoid.

screenshot from 2019-02-22 11-34-33

### CODE REVIEW

**Comment by mpowelson** [comment]
@Levi-Armstrong

File: exercises/Optimization_Based_Planning/solution_ws/src/.rosinstall [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- git: {local-name: tesseract, uri: 'https://github.com/ros-industrial-consortium/trajopt_ros.git', version: kinetic-devel}
```
Comment (gavanderhoorn) [inline exercises/Optimization_Based_Planning/solution_ws/src/.rosinstall]:
```suggestion
- git: {local-name: tesseract, uri: 'https://github.com/ros-industrial-consortium/tesseract.git', version: kinetic-devel}
```

File: exercises/Optimization_Based_Planning/solution_ws/src/.rosinstall [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+- git: {local-name: trajopt, uri: 'https://github.com/ros-industrial-consortium/trajopt_ros.git', version: kinetic-devel}
```
Comment (gavanderhoorn) [inline exercises/Optimization_Based_Planning/solution_ws/src/.rosinstall]: This points to branch tips, which can be moving targets. Should this point to specific tags or commits instead?

Not sure how much we value stability, but would be a shame if you spend 30 mins trying to help students fix some error which was introduced after you've verified that "everything worked" when you wrote these exercises.

Comment (mpowelson) [reply exercises/Optimization_Based_Planning/solution_ws/src/.rosinstall]: That's a good point. I just tied it to the latest commits in Tesseract, Trajopt, and iiwa_stack that are known to work for Kinetic and Melodic

**Pull request merged**